### PR TITLE
Move check for lossy whole-number floats out of `excessive_precision`

### LINKED ISSUE #5201: Broken suggestion from excessive_precision lint
This is minimized from some code in serde_json: https://github.com/serde-rs/json/blob/ed479b4656be48760f3d3373d0fe476541e94e0e/src/de.rs#L1025

```rust
fn main() {
    let _ = 1e099;
}
```

```console
error: literal cannot be represented as the underlying type without loss of precision
 --> src/main.rs:2:13
  |
2 |     let _ = 1e099;
  |             ^^^^^ help: consider changing the type or replacing it with: `1e9_9.0`
  |
  = note: `#[deny(clippy::excessive_precision)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#excessive_precision
```

Clippy's suggestion of writing this as `1e9_9.0` seems wrong and does not work.

```console
error[E0610]: `{float}` is a primitive type and therefore doesn't have fields
 --> src/main.rs:2:19
  |
2 |     let _ = 1e9_9.0;
  |                   ^
```

Separately, I don't feel that there should be an on-by-default correctness lint against writing 1e099. I understand that this number can't be represented exactly by f64 (the value ends up being 999999999999999967336168804116691273849533185806555472917961779471295845921727862608739868455469056.0) but this seems excessively noisy for a default lint.

Mentioning @krishna-veerareddy @flip1995 because this lint was last touched in #5185.

*clippy 0.0.212 (2855b21 2020-02-19)*

### PR DESCRIPTION
**Pull request by krishna-veerareddy**
changelog: Add new lint `lossy_float_literal` to detect lossy whole number float literals and move it out of `excessive_precision` again.

Fixes #5201

### CODE REVIEW

File: tests/ui/lossy_float_literal.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#![warn(clippy::lossy_float_literal)]
```
Comment (flip1995) [inline tests/ui/lossy_float_literal.rs]:
```suggestion
// run-rustfix
#![warn(clippy::lossy_float_literal)]
```

File: clippy_lints/src/lossy_float_literal.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    ///
+    /// ```rust
+    /// // Bad
+    /// let _: f32 = 16_777_217.0; // 16_777_216.0
+    ///
+    /// // Good
+    /// let _: f32 = 16_777_216.0;
+    /// let _: f64 = 16_777_217.0;
+    /// ```
+    pub LOSSY_FLOAT_LITERAL,
```
Comment (flip1995) [inline clippy_lints/src/lossy_float_literal.rs]: I'd put this in the same LintPass as the `excessive_precission` lint, to share some code. You can then maybe change the name of the `excessive_precision.rs` module to `float_literals.rs`.

File: clippy_lints/src/excessive_precision.rs [hunk 3]
```diff
@@ -6,31 +6,33 @@ use rustc_errors::Applicability;
...
     ///
     /// // Good
-    /// let a: f64 = 0.123_456_789_9;
-    /// let b: f64 = 16_777_216.0;
+    /// let v: f64 = 0.123_456_789_9;
+    /// println!("{}", v); //  0.123_456_789_9
     /// ```
     pub EXCESSIVE_PRECISION,
-    correctness,
+    style,
```
Comment (flip1995) [inline clippy_lints/src/excessive_precision.rs]: Yeah, let's leave it in `style`. deny-by-default is probably to harsh for this lint.

File: clippy_lints/src/lossy_float_literal.rs (hunk 2 above)
Comment (krishna-veerareddy) [reply clippy_lints/src/lossy_float_literal.rs]: Makes sense. Will do.

File: clippy_lints/src/excessive_precision.rs (hunk 3 above)
Comment (krishna-veerareddy) [reply clippy_lints/src/excessive_precision.rs]: Ok sounds good. `style` just doesn't feel like the right place to put it but we can revisit this later.

**Pull request merged**